smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatEntityIntegrationTest.java

```diff
@@ -0,0 +1,204 @@
+/**
+ *
+ * Copyright 2021 Dan Caseley
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jivesoftware.smackx.muc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Map;
+
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo;
+import org.jivesoftware.smackx.disco.packet.DiscoverItems;
+
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.igniterealtime.smack.inttest.TestNotPossibleException;
+import org.igniterealtime.smack.inttest.annotations.SmackIntegrationTest;
+import org.jxmpp.jid.EntityBareJid;
+import org.jxmpp.jid.impl.JidCreate;
+import org.jxmpp.jid.parts.Resourcepart;
+
+public class MultiUserChatEntityIntegrationTest extends AbstractMultiUserChatIntegrationTest {
+
+    public MultiUserChatEntityIntegrationTest(SmackIntegrationTestEnvironment environment)
+                    throws SmackException.NoResponseException, XMPPException.XMPPErrorException,
+                    SmackException.NotConnectedException, InterruptedException, TestNotPossibleException {
+        super(environment);
+    }
+
+    /**
+     * Asserts that a MUC service can be discovered
+     *
+     * <p>From XEP-0045 § 6:</p>
+     * <blockquote>
+     * A MUC implementation MUST support Service Discovery (XEP-0030) [9] ("disco").
+     * </blockquote>
+     *
+     * <p>From XEP-0045 § 6.1:</p>
+     * <blockquote>
+     * An entity often discovers a MUC service by sending a Service Discovery items ("disco#items") request to its own
+     * server. The server then returns the services that are associated with it.
+     * </blockquote>
+     *
+     * @throws Exception when errors occur
+     */
+    @SmackIntegrationTest
+    public void mucTestForDiscoveringMUC() throws Exception {
+        DiscoverItems items = ServiceDiscoveryManager.getInstanceFor(conOne).discoverItems(conOne.getXMPPServiceDomain());
+        assertTrue(items.getItems().stream().anyMatch(i -> i.getEntityID().toString().contains("conference"))); //TODO remove magic string, without making it disco=disco.
```

Comment:
I think this would be more readable if the stream of Items would be mapped to a stream of strings. Then the last `anyMatch()` simply becomes `anyMatch(s -> s.contains("conference")`.

Also nit: space after `//`. And transform this to an "above the line" comment. Github's UI actually gives a good motivation for this transformation, as I currently can't read the whole comment.